### PR TITLE
OWPreprocess: Add information on tokens

### LINKED ISSUE #670: Preprocess Text: Add info on the number of output tokens
**Is your feature request related to a problem? Please describe.**
Preprocess Text used to report the number of output tokens. It no longer does.

**Describe the solution you'd like**
Add the number of output tokens and types to the widget. The info can be added to either the status bar (output section) or directly to the widget under the Preview.

**Describe alternatives you've considered**
/

**Additional context**
/

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Fixes #670.

##### Description of changes
Add a box with counts of tokens and types.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/text/widgets/tests/test_owpreprocess.py [hunk 1]
```diff
@@ -27,6 +27,16 @@ def test_outputs(self):
         self.assertIsNot(pp_data, self.corpus)
         self.assertNotEqual(pp_data, self.corpus)
 
+    def test_previews(self):
+        self.send_signal(self.widget.Inputs.corpus, self.corpus)
+        self.get_output(self.widget.Outputs.corpus)
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/tests/test_owpreprocess.py:32]: This doesn't look too elegant, but it does assure the output. Any nicer ways to do it?

File: orangecontrib/text/widgets/owpreprocess.py [hunk 2]
```diff
@@ -960,6 +960,10 @@ def __init__(self):
         self.preview = ""
         gui.label(box, self, "%(preview)s", wordWrap=True)
         self.controlArea.layout().insertWidget(1, box)
+        box = gui.vBox(self.controlArea, "Output")
+        self.output_info = ""
+        gui.label(box, self, "%(output_info)s", wordWrap=True)
+        self.controlArea.layout().insertWidget(2, box)
         self.controlArea.setFixedWidth(220)
```
Comment (VesnaT) [inline orangecontrib/text/widgets/owpreprocess.py:967]: While you're at it, please remove this line. The control area will look a lot better.

File: orangecontrib/text/widgets/owpreprocess.py [hunk 3]
```diff
@@ -960,6 +960,10 @@ def __init__(self):
         self.preview = ""
         gui.label(box, self, "%(preview)s", wordWrap=True)
         self.controlArea.layout().insertWidget(1, box)
+        box = gui.vBox(self.controlArea, "Output")
```
Comment (VesnaT) [inline orangecontrib/text/widgets/owpreprocess.py:963]: I think we once agreed to put the input-unrelated info into the buttonsArea.

image

If you prefer it your way, feel free to do it like that.
If not, here are some hints:
- set `buttons_area_orientation = Qt.Vertical`
- replace `box = gui.vBox(self.controlArea, "Output")` with `box = gui.vBox(self.buttonsArea, "Output")`
- replace `self.controlArea.layout().insertWidget(2, box)` with `self.buttonsArea.layout().insertWidget(0, box)`

File: orangecontrib/text/widgets/tests/test_owpreprocess.py [hunk 4]
```diff
@@ -27,6 +27,16 @@ def test_outputs(self):
...
+        self.assertTrue(self.widget.preview)
+        self.assertTrue(self.widget.output_info)
+        self.send_signal(self.widget.Inputs.corpus, None)
+        self.get_output(self.widget.Outputs.corpus)
```
Comment (VesnaT) [inline orangecontrib/text/widgets/tests/test_owpreprocess.py:36]: Same as above.

**Comment by ajdapretnar** [comment]
@VesnaT I addressed all your comments. Thank you for the hint on buttonsArea, it looks so much better!

**Pull request merged**